app/models/data_update_script.rb

```diff
@@ -0,0 +1,39 @@
+class DataUpdateScript < ApplicationRecord
+  DIRECTORY = Rails.root.join("lib/data_update_scripts").freeze
+  NAMESPACE = "DataUpdateScripts".freeze
+  STATUSES = %i[enqueued working succeeded failed].freeze
+
+  enum status: STATUSES
```

Comment:
One thing I always encourage people to be careful about when using Rails' enums, is that array order affects integer values stored in the DB. So if someone reorders `STATUSES` by e.g. adding a new one in the middle, things will return a different status than before. The `enum` helper also excepts a hash, so I got into the habit of using that instead of arrays:

```ruby
STATUSES = { enqueued: 0, working: 1, succeeded: 2, failed: 3 }
```

Reply:
Interesting, I have never run into that problem but better safe than sorry!